rust-toolchain

```diff
@@ -1 +1 @@
-stable-2021-11-01
+nightly
```

Comment:
There's only one thing to take into account if we go for that, and it's the following:

We should stick to a specific nightly version for which `rustfmt` exists and works. Since from time to time there are some  nightly versions that come without `rustfmt` component and that would break the CI.

My suggestion is to use the latest nightly which already contains the feature needed by @ChihChengLiang and also the bug-fix for #178 specifying also the date.

In this case: `2021-11-17` which corresponds to `nightly-1.58`